### PR TITLE
8278796: Incorrect behavior of FloatVector.withLane on X86

### PR DESCRIPTION
**Pull request by jatin-bhateja**
- Incorrect operand is being passed to insertps instruction which causes incorrectness issues in FloatVector.withLane operation.
 - Existing JTREG test cases have been modified appropriately with a non-zero insertion index.

Kindly review and share your comments.
Best Regards,
Jatin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8278796](https://bugs.openjdk.java.net/browse/JDK-8278796): Incorrect behavior of FloatVector.withLane on X86

### Reviewers
 * [Sandhya Viswanathan](https://openjdk.java.net/census#sviswanathan) (@sviswa7 - **Reviewer**) ⚠️ Review applies to eac1660eeabb218ec88db02d1ad96e57c1bacf02
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk18 pull/28/head:pull/28` \
`$ git checkout pull/28`

Update a local copy of the PR: \
`$ git checkout pull/28` \
`$ git pull https://git.openjdk.java.net/jdk18 pull/28/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 28`

View PR using the GUI difftool: \
`$ git pr show -t 28`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk18/pull/28.diff">https://git.openjdk.java.net/jdk18/pull/28.diff</a>

</details>

### CODE REVIEW

**Comment by sviswa7** [comment]
@jatin-bhateja Thanks for fixing this issue. The patch looks good to me.

**Review by sviswa7 — APPROVED** [review]

**Review by vnkozlov — COMMENTED** [review]
Looks fine to me too. I will run testing.

@jatin-bhateja please, update Description in bug report (it is empty now).

**Review by vnkozlov — APPROVED** [review]
Testing passed.

**Review by PaulSandoz — COMMENTED** [review]
The changes to the tests look good, ideally we should test over all lane indexes, but i believe the insert intrinsic currently requires the lane index be a constant. Unsure if that is a restriction that can be lifted.

**Comment by jatin-bhateja** [comment]
> The changes to the tests look good, ideally we should test over all lane indexes, but i believe the insert intrinsic currently requires the lane index be a constant. Unsure if that is a restriction that can be lifted.

Hi @PaulSandoz , Your comments have been addressed.

**Comment by PaulSandoz** [comment]
I think that is a little more robust, API-wise, but do i misunderstand the insert intrinsic with regards to requiring the lane index be a constant?

**Comment by sviswa7** [comment]
@PaulSandoz  The withLane implementation calls the withLaneHelper with constant index. e.g. please see Int128Vector withLane implementation has a switch statement to achieve this.

**Comment by PaulSandoz** [comment]
> @PaulSandoz The withLane implementation calls the withLaneHelper with constant index. e.g. please see Int128Vector withLane implementation has a switch statement to achieve this.

Oh yes, of course, thanks!

**Review by vnkozlov — COMMENTED** [review]
I have to retest before update approval.

**Comment by PaulSandoz** [comment]
Testing passed on latest commit.

**Review by vnkozlov — APPROVED** [review]
Good.

**Pull request closed without merging**
